File: articles/systems-thinking-patterns-interdependence-structural-change/c/stock_flow_model.c
#include <stdio.h>

int main(void) {
    const int steps = 20;
    double stock[steps];
    double inflow = 5.0;
    double outflow = 3.5;

    stock[0] = 50.0;

    for (int t = 1; t < steps; t++) {
        stock[t] = stock[t - 1] + inflow - outflow;
        if (stock[t] < 0.0) {
            stock[t] = 0.0;
        }
    }

    printf("Stock-flow simulation\n");

    for (int t = 0; t < steps; t++) {
        printf("time=%d, stock=%.2f\n", t, stock[t]);
    }

    return 0;
}

File: articles/systems-thinking-patterns-interdependence-structural-change/fortran/stock_flow_model.f90
program stock_flow_model
  implicit none

  integer, parameter :: steps = 20
  real :: stock(steps)
  real :: inflow, outflow
  integer :: t

  stock(1) = 50.0
  inflow = 5.0
  outflow = 3.5

  do t = 2, steps
     stock(t) = max(0.0, stock(t - 1) + inflow - outflow)
  end do

  print *, "Stock-flow simulation"

  do t = 1, steps
     print *, t, stock(t)
  end do

end program stock_flow_model
